contracts/Library/ClaimLib.sol

```diff
@@ -0,0 +1,76 @@
+pragma solidity ^0.4.18;
+
+import 'contracts/zeppelin/SafeMath.sol';
+
+library ClaimLib {
+    using SafeMath for uint;
+
+    struct ClaimData {
+        address claimedBy;          // The address that has claimed the txRequest.
+
+        uint claimDeposit;          // The deposit amount that was put down by the claimer.
+
+        uint requiredDeposit;       // The required deposit to claim the txRequest.
+
+        uint8 paymentModifier;      // An integer constrained between 0-100 that will be applied to the
+                                    // request payment as a percentage.
+    }
+
+    /*
+     * @dev Mark the request as being claimed.
+     * @param self The ClaimData that is being accessed.
+     * @param paymentModifier The payment modifier.
+     */
+    function claim(
+        ClaimData storage self, 
+        uint8 _paymentModifier
+    ) 
+        internal returns (bool)
+    {
+        self.claimedBy = msg.sender;
+        self.claimDeposit = msg.value;
+        self.paymentModifier = _paymentModifier;
+    }
+
+    /*
+     * Helper: returns whether this request is claimed.
+     */
+    function isClaimed(ClaimData storage self) 
+        internal view returns (bool)
+    {
+        return self.claimedBy != 0x0;
+    }
+
+    /*
+     * @dev Calculates the amount that must be supplied as a deposit to claim.  
+     * This is set to the maximum possible payment value that could be paid out by this request.
+     */
+
+    // TODO: add `requiredDeposit` and remove the hard-coding of the `2 *
```

Comment:
This `TODO` looks like this was not addressed.  The idea was to remove the hard-coded `2 * payment` and instead add an additional argument to scheduling which was the required claim deposit.

* Add `uint requiredDeposit` to the `ClaimData` struct.
* Change the checks to use `claimData.requiredDeposit` instead of `2 * payment`

Reply:
Ok

Reply:
Do I understand correctly that this should be added to the uintArgs on the `schedule()` API? So we bump up the uint[7] array used now to an uint[8]. 

```
function schedule(address   _toAddress,
                      bytes     _callData,
                      uint[8]   _uintArgs)
        doReset
        public payable returns (address);
```

I've made this modification in the contracts and I'm reworking the tests that used the old API to reflect this change.

Reply:
```
        futureTransaction.toAddress         = _toAddress;
        futureTransaction.callData          = _callData;
        futureTransaction.callGas           = _uintArgs[0];
        futureTransaction.callValue         = _uintArgs[1];
        futureTransaction.windowSize        = _uintArgs[2];
        futureTransaction.windowStart       = _uintArgs[3];
        futureTransaction.gasPrice          = _uintArgs[4];
        futureTransaction.donation          = _uintArgs[5];
        futureTransaction.payment           = _uintArgs[6];
        futureTransaction.requiredDeposit   = _uintArgs[7];
```

Reply:
Yes, this is correct.